fabric/operations.py

```diff
@@ -1198,8 +1198,14 @@ def local(command, capture=False, shell=None):
         The return value attributes ``.command`` and ``.real_command``.
     """
     given_command = command
-    # Apply cd(), path() etc
+    # Apply cd(), path() etc, and set the SHELL env var to be consistent
+    set_shell = env.shell_env.get('SHELL')
+    env.shell_env['SHELL'] = shell or '/bin/sh'
     with_env = _prefix_env_vars(command, local=True)
+    if set_shell is None:
+        del env.shell_env['SHELL']
+    else:
+        env.shell_env['SHELL'] = set_shell
```

Comment:
This is messy and unnecessary. Just copy the env, modify it, and pass it to `Popen()`:

```python
sub_env = os.environ.copy()
sub_env['SHELL'] = shell or '/bin/sh'
...
subprocess.Popen(cmd_arg, env=sub_env, shell=True, ...)
```

I still think just deleting SHELL from `sub_env` would be simpler, as it gets the actual shell to set the appropriate value, as bash does. I don't know if fabric really runs on windows (I think it theoretically does), but that would be an example of a case where `Popen()` would not actually use /bin/sh so assuming it does here would be wrong.

Reply:
Definitely unnecessarily messy, not sure what I was thinking. Will fix that.

I tried the approach of deleting SHELL from sub_env, but that doesn't actually seem to solve the problem. In my case, `local('echo $SHELL')` still returns `/usr/local/bin/fish`.

Windows not working with this is a good point.

Reply:
... might be because you tried with `env.shell_env` which is an additive-only fabric thing.

Anyway, thanks for the report and the PR, it's a good find :)